### PR TITLE
- FemtoDream : Adds TOF Mass Cut facility to study Sideband correlation

### PR DESCRIPTION
**Pull request by sbhawani**

### CODE REVIEW

File: PWGCF/FEMTOSCOPY/FemtoDream/AliFemtoDreamTrackCuts.cxx [hunk 1]
```diff
@@ -69,6 +70,12 @@ AliFemtoDreamTrackCuts::AliFemtoDreamTrackCuts()
       fNSigValueITS(3.),
       fPIDPTPCThreshold(0),
       fPIDPITSThreshold(0),
+      fTOFInMassCut(false),
+      fCutArroundPeakTOFInvMass(false),
+      fCutTOFInvMassSidebands(false),
+      fInvMassCutWidth(0),
+      fInvMassCutSBdown(0),
+      fInvMassCutSBup(0),
```
Comment (sartozza) [inline PWGCF/FEMTOSCOPY/FemtoDream/AliFemtoDreamTrackCuts.cxx:78]: please put fInvTOFMass... in all names in order to understand is related to TOF.

File: PWGCF/FEMTOSCOPY/FemtoDream/AliFemtoDreamTrackCuts.cxx [hunk 2]
```diff
@@ -133,6 +141,12 @@ AliFemtoDreamTrackCuts::AliFemtoDreamTrackCuts(
       fNSigValueITS(cuts.fNSigValueITS),
       fPIDPTPCThreshold(cuts.fPIDPTPCThreshold),
       fPIDPITSThreshold(cuts.fPIDPITSThreshold),
+      fTOFInMassCut(cuts.fTOFInMassCut),
+      fCutArroundPeakTOFInvMass(cuts.fCutArroundPeakTOFInvMass),
+      fCutTOFInvMassSidebands(cuts.fCutTOFInvMassSidebands),
+      fInvMassCutWidth(cuts.fInvMassCutWidth),
+      fInvMassCutSBdown(cuts.fInvMassCutSBdown),
+      fInvMassCutSBup(cuts.fInvMassCutSBup),
```
Comment (sartozza) [inline PWGCF/FEMTOSCOPY/FemtoDream/AliFemtoDreamTrackCuts.cxx:149]: Put TOF in the name

File: PWGCF/FEMTOSCOPY/FemtoDream/AliFemtoDreamTrackHist.h [hunk 3]
```diff
@@ -18,7 +18,7 @@
 class AliFemtoDreamTrackHist {
  public:
   AliFemtoDreamTrackHist();
-  AliFemtoDreamTrackHist(bool DCADist, bool CombSig, bool TOFM, float pTmin = 0.5, float pTmax = 4.05,  int MultRangeLow = 27, int MultRangeHigh = 55);
+  AliFemtoDreamTrackHist(bool DCADist, bool CombSig, bool TOFM,bool TOFMSq, float pTmin = 0.5, float pTmax = 4.05,  int MultRangeLow = 27, int MultRangeHigh = 55);
```
Comment (sartozza) [inline PWGCF/FEMTOSCOPY/FemtoDream/AliFemtoDreamTrackHist.h:21]: Here please put the bool TOFMSq as last argument and put it to false, for backward compatibility

File: PWGCF/FEMTOSCOPY/FemtoDream/AliFemtoDreamTrackCuts.cxx [hunk 4]
```diff
@@ -564,6 +590,44 @@ bool AliFemtoDreamTrackCuts::PIDCuts(AliFemtoDreamTrack *Track) {
   return pass;
 }
 
+bool AliFemtoDreamTrackCuts::TOFMassCuts(AliFemtoDreamTrack *Track) {
```
Comment (sartozza) [inline PWGCF/FEMTOSCOPY/FemtoDream/AliFemtoDreamTrackCuts.cxx:593]: The function implemented like this makes the PID part very messy.
So we propose as a solution to include this invariant mass TOF cut inside the PIDCuts function.
Please do this by inserting an option (enumerator or bool) to select whether to use TOFnSigma or InvMassTOF.
You can then re-use your function TOFMassCuts and put the lines of the code around lines 545, so where we have the TOF PID part.

**Review by sartozza — CHANGES_REQUESTED** [review]

**Review by dmuhlhei — CHANGES_REQUESTED** [review]
Are the changes in AddTask_GammaConvV1 really intended?

**Comment by sbhawani** [comment]
Dear all,

This is a mistake, I am fixing. The changes in AddTask_GammaConvV1 are not
intentional. Please *do not *approve!
Apologies for inconvenience!

**Pull request closed without merging**